### DISCUSSION TITLE
How to get started?

### DISCUSSION THREAD

**Discussion by rsoika** [tiredofit/docker-collabora-online#14] [Q&A]
Hi,
I am new to libreoffice-online and currently a little bit lost.

We are developing the Open Source project [Imixs-Documents](https://github.com/imixs/imixs-documents). This is a Workflow Management System for small and medium sized enterprises and organisations. We are planning to provide a way to edit office documents - with libreoffice-online. 

I used your container and the example docker-compose.yaml file to get started.

But to be honest, I don't understand the concept of Libreoffice at the moment. What exactly do I need to do to open or edit an Office Document?

Our [workflow engine](https://www.imixs.org) runs on the Jakarta EE Stack - so we have a lot of possibilities to call services and exchange data. 

Maybe someone can give me an short introduction to what exactly the server expects after I started it.

Thanks a lot in advance

===
Ralph

**Comment by tiredofit** [comment]
Sweet! I love BPMN and workflow!
I'm just a lowly builder of this image, and not affiliated with LibreOffice, however I have integrated this into applications, or at least my developers have. 

Have a peek at the WOPI protocol https://wopi.readthedocs.io/en/latest/ and there's quite a bit of info in the LibreOffice Online repository: https://github.com/LibreOffice/online/blob/master/wsd/README.

Hopefully this helps? Will watch your solution closely :)

**Reply by rsoika** [reply]
Ok, thanks for your reply. I understand little by little that a separate microservice, implementing the WOPI protocol, is needed to get things working. I hoped that there is some trick so I can have a first look at the libreoffice-online web editor before doing this implementation. Maybe I will write a tutorial during the implementation of our integration service.

**Reply by tiredofit** [reply]
Fully understood - What you could do is install a quick copy of Nextcloud (I make one thats fairly straight forward `tiredofit/nextcloud`) and then head into their "Apps" section and install "Rich Documents" or as they may also call it "CODE". When you load it the settings will ask which server you wish to use. You could use this image, or use their demo servers to kick the tires.

There also may be another integration possibility for you - OnlyOffice Editors are fairly strong, have great compatibility with the MS documents and operate in a similar fashion. I believe Onlyoffice is Open source as well however I am not sure off the top of my head about the license. You can also evaluate that editor within Nextcloud.

**Reply by rsoika** [reply]
I have written a blog post about how to implement the WOPI Host to integrate libreoffice online. 
https://ralph.blog.imixs.com/2021/05/07/libreoffice-online-how-to-integrate/

....I learned a lot ;-)

**Reply by tiredofit** [reply]
I'd say - that's the missing Quick start developers guide right there. I'll be watching your work (and blog) in future to see how your workflow solution shapes up!
